### ISSUE TITLE
Magento removes end tag

### ISSUE THREAD

**Issue by Eddcapone** [magento/magento2#31958]
### Preconditions (*)

1. 2.4.1
2. Linux Server (we are hosted on Rackspeed)
3. Apache Webserver

### Steps to reproduce (*)

1. Create block:

**app\code\Company\Base\Block\Scripts\Filter.php**

```
<?php

namespace Company\Base\Block\Scripts;

class Filter extends \Magento\Framework\View\Element\Template
{   
    public function __construct(
        \Magento\Framework\View\Element\Template\Context $context,
        array $data = []
    ) {          
        parent::__construct($context, $data);
    }



}

```

2. Create Template:

**app\code\Company\Base\view\frontend\templates\scripts\filter.phtml**

```
<script>
require([
    'jquery',
    'domReady!'
], function ($) {
    var x = "test";
    console.log("<li id='option_"+ x + "' data-target='"+ x +"'>" + x + "</li>");
});

</script>
```

3. Create new CMS Site with following content:

{{block class="Company\\Base\\Block\\Scripts\\Filter" template="Company_Base::scripts/filter.phtml"}}

4. Load the CMS site in the frontend
5. Open developer console and expect console output.

### Expected result (*)

Console output should be:

`<li id='option_test' data-target='test'>test</li>`

### Actual result (*)

Console output is:

`<li id='option_test' data-target='test'>test`

↑ It misses the end `</li>` tag.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by ihor-sviziev** [comment]
@Eddcapone do you have the same issue with default template class?

**Comment by Eddcapone** [comment]
@ihor-sviziev I just tested it again on a fresh Magento 2 (with sample data) and I get the same bug. As you can see the custom class, has nothing special in it, just a constructor. There is no plugins, since this is a fresh magento 2.

**Comment by Eddcapone** [comment]
@ihor-sviziev Oh, and I am using Magento 2 Enterprise, not sure if it makes a difference.

**Comment by Eddcapone** [comment]
I figured out, that you can write it as this as a workaround:

`console.log("<li id='option_"+ x + "' data-target='"+ x +"'>" + x + "<\/li>");`

Write `"<\/li>"` instead of` `"</li>"

I hope this gets fixed soon, since it causes many bugs which are hard to find, especially if you dont know this bug.

**Comment by ihor-sviziev** [comment]
@sidolov @sivaschenko seems like this should be P1. Could you check it?

**Comment by engcom-Alfa** [comment]
Hi, @Eddcapone.
Unfortunately, we are not able to reproduce the issue on a fresh `2.4-develop CE`, `2.4-develop EE `, and `2.4.1 composer.`
Everything works as expected.

_**Preconditions:**_

- Php 7.4
- Apache
- Mysql 8.0

_**Manual testing scenario:**_

1. Create a custom module:

- Create block:
**app\code\Custom\Module\Block\Scripts\Filter.php**

```php
<?php

namespace Custom\Module\Block\Scripts;

class Filter extends \Magento\Framework\View\Element\Template
{   
    public function __construct(
        \Magento\Framework\View\Element\Template\Context $context,
        array $data = []
    ) {          
        parent::__construct($context, $data);
    }
}
```
- Create Template:
**app\code\Custom\Module\view\frontend\templates\scripts\filter.phtml**

```js
<script>
require([
    'jquery',
    'domReady!'
], function ($) {
    var x = "test";
    console.log("<li id='option_"+ x + "' data-target='"+ x +"'>" + x + "</li>");
});

</script>
```
2. Create new CMS Site with following content:

{{block class="Custom\\Module\\Block\\Scripts\\Filter" template="Custom_Module::scripts/filter.phtml"}}

3. Load the CMS site in the frontend
4. Open developer console and inspect console output.

**Actual Result:** :heavy_check_mark: Console output is:
```js
<li id='option_test' data-target='test'>test</li>
```
![2021-02-04_09-48](https://user-images.githubusercontent.com/51679138/106864088-17006800-66d2-11eb-942f-6e17d6a4ba23.png)

@Eddcapone Could you take a look?
Thanks!

**Comment by Eddcapone** [comment]
@engcom-Alfa This is weird. I have no explanation why it works on your side but not on mine.
I just tested it with Firefox and Chrome in inspector mode and I get the same bug again and again.

Does anyone have an idea what might cause it?

**Comment by ihor-sviziev** [comment]
@Eddcapone can we add it to some cms page and test on test instances?

**Comment by Eddcapone** [comment]
@ihor-sviziev Good idea, maybe this will identify the problem. It might be a server problem.

**Comment by Eddcapone** [comment]
If I just execute it in the browser, then It works. So the server does fiddle with the result for unknown reasons.

![image](https://user-images.githubusercontent.com/16349349/106873583-e79f2a80-66d4-11eb-94b7-6fe9730cda77.png)

**Comment by Eddcapone** [comment]
We can also simplify the example to this:

```
<script>
require([
    'jquery',
    'domReady!'
], function ($) {

    console.log("<li>test</li>");
});

</script>
```

**Output**: `<li>test`

**Comment by ihor-sviziev** [comment]
@magento give me 2.4-develop instance

**Comment by ihor-sviziev** [comment]
Hi @Eddcapone,
I wasn't able to reproduce this issue on test instance:
https://beb9d6c6fa8bee902aff6686df4369d6-2-4-develop.instances.magento-community.engineering/
![image](https://user-images.githubusercontent.com/1873745/106883234-65b8fc80-66e8-11eb-9b3a-1d8e2782941d.png)
![image](https://user-images.githubusercontent.com/1873745/106883152-47eb9780-66e8-11eb-9cd7-ef076d9918b4.png)

**Comment by Eddcapone** [comment]
@ihor-sviziev , Thanks for testing it. I also tested it on another computer and get the same bug, so it must be definitely a serverside problem.  Does anyone have an idea why the bug exist on our server and not on the test server from magento?

**Comment by Eddcapone** [comment]
@ihor-sviziev I added the block to a Page Builder HTML Node, maybe that is causing it? I can't test on a normal open source magento.

**Comment by ihor-sviziev** [comment]
@engcom-Alfa could you double check that on EE version with a page builder ?

**Comment by Eddcapone** [comment]
@ihor-sviziev The bug must have something to do with the page builder. I just tested it again by adding my script to the page.wrapper container via the default.xml layout file and it works there.

**Comment by engcom-Alfa** [comment]
The issue is reproducible on the EE 2.4.1 with a page builder.
![2021-02-05_13-09](https://user-images.githubusercontent.com/51679138/107026525-7edaaf80-67b3-11eb-8854-51cc5e8b3fd5.png)
It looks like the problem may be related to Page Builder functionality.
